src/app/rpmostree-clientlib.cxx

```diff
@@ -1782,3 +1787,10 @@ error_if_driver_registered (GBusType          bus_type,
 
   return TRUE;
 }
+
+gboolean
+rpmostree_client_reboot (GError ** error)
+{
+  execlp ("reboot", "reboot", NULL);
```

Comment:
Is this `exec` really intended here? I was somehow expecting some kind of fork+exec spawning (like in the original `rpmostreed_reboot`). This way instead it looks a bit like a dangerous landmine, because in the non-error case any other code following this function call won't really be reached.

Reply:
In a non-error case, the process will get killed by `SIGTERM` at some point soon later, so any code after would be inherently racy.

Reply:
Self-note, I had a personal doubt related to blocking-behavior of this action, and I double-checked the systemd manual:
> This command is asynchronous; it will return after the reboot operation is enqueued, without waiting for it to complete.

Reply:
Compare https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.exec

Reply:
Brutal but fair point. Although Rust type-system represents such kind of never-returning functions, while here it's one more thing to be keep in mind. At least a comment would help to highlight the intentional blackhole.

Anyway I agree this way it makes potential races obvious upfront, and we don't seem to currently have any critical logic running afterwards, so fine for me.

Reply:
Well it *can* return, just on error.  This is where the Rust type system shows its strength because unlike almost every other function it doesn't return `Result<()>` it returns `Error`.

Reply:
Now you're completely right that the command itself is async...which is not usually what one wants.  But that's also not a new issue here.  The bug we're fixing is that the client can sometimes spuriously exit with an error because the daemon gets killed before it can send the success reply.

Reply:
"[never](https://doc.rust-lang.org/std/primitive.never.html)" as in the underlying diverging a case which can't be represented here. Sorry English language is confusing :)

Anyway I think we are in agreement. Mind adding a docstring highlighting that the `TRUE` case does not really return from this function blackhole and the whole execution flow diverges instead, and then we land it?